docs/proposals/rollout/rollout.md

```diff
@@ -0,0 +1,531 @@
+
+### Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+In this section, we'll dive into the detailed API design for the Unified Rollout Feature.
+
+These APIs are designed to facilitate Kurator's integration with Flagger to enable the required functionality.
+
+Unlike Flagger, we may need to adjust Unified Rollout to reflect our new strategy and decisions.
+
+#### Unified Rollout API
+
+Kurator is designed to unify the installation of Flagger as a fleet plugin in a given single or multiple clusters.
+
+Then use the Kurator application to distribute the Flagger configuration. Kurator's unified configuration distribution.
+
+Kurator puts the Rollout's api under the [Application](https://github.com/kurator-dev/kurator/blob/main/pkg/apis/apps/v1alpha1/types.go) CRD, so that when Kurator deploys the workload in the target cluster, it also deploys the corresponding Rollout policy.
+
+Here's the preliminary design for the Unified Rollout:
+
+```console
+// ApplicationSyncPolicy defines the configuration to sync an artifact.
+// Only oneof `kustomization` or `helm` can be specified to manage application sync.
+// ApplicationSyncPolicy distributes the Rollout configuration 
+// at the same time as the application deployment, if needed. 
+type ApplicationSyncPolicy struct {
+    // Rollout defines the rollout Configurations to be used.
+    // If specified, a uniform Rollout policy is configured for this installed object.
+    // +optional
+    Rollout *RolloutConfig `json:"rolloutPolicy,omitempty"`
+}
+
+type RolloutConfig struct {
+    // Testloader defines Whether to install testloader for users. Default is true.
+    // Testloader generates traffic during canary analysis.
+    // If set it to false, user need to install the testloader himself.
+    // If set it to true or leave it blank, Kurator will install the flagger's testloader.
+    // +optional
+    TestLoader bool `json:"testLoader,omitempty"`
+
+    // Kurator only supports istio for now.
+    // New Provider will be added later.
+    // +optional
+    TrafficRoutingProvider string `json:"trafficRoutingProvider,omitempty"`
+
+    // Workload specifies what workload to deploy the test to. 
+    // Workload of type deployment or daemonSet.
+    Workload *WorkloadReference `json:"workload"`
+
+    // ServiceName holds the name of a service which matches the `Workload`.
+    ServiceName string `json:"serviceName"`
+
+    // Port of the workload's Service which traffic access.
```

Comment:
```suggestion
    // Port exposed by the service.
```

Reply:
What happen if this port does not match any ports of the service named by `ServiceName`. Or it is not a HTTP port